internal/export/distro/messaging.go

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package distro
+
+import (
+	"github.com/edgexfoundry/go-mod-messaging/pkg/messaging"
+	"github.com/edgexfoundry/go-mod-messaging/pkg/types"
+)
+
+func InitMessaging(messageErrors chan error, messages chan *types.MessageEnvelope) {
+	// Create the messaging client
```

Comment:
I don't see that the channels initialized here are explicitly closed anywhere, for example when the service is stopped. Can that be added to distro.Destruct()? The best approach might be to move the explicit initialization out of main.go -- talking about this
```
errs := make(chan error, 2)
eventCh := make(chan *types.MessageEnvelope, 10)
```

Reply:
it may be more consistent to init, use and close it in the same place. Rather than init and close it in one file, but use it in another. thoughts?

Reply:
Well, the whole of export distro, other than `main.go`, is one package. You'll see an init.go file in `internal/export/distro` that handles service initialization. If you use the same approach for the messaging channels that the Registry integration took, then you'll be able to close the channels explicitly.

Also, as it is now, there's probably a subtle bug in `main.go` when re-using the `errs` channel for messaging or distro loop errors. That's really intended for an interrupt from the OS. It could be that we'd prefer to log an error from distro or the messaging pub/sub rather than have the service quit. 

- You could then call `distro.initMessaging()` from init.go::init() 
- modify lines 65, 66 in `main.go` to just be:
distro.Loop()

If you get stuck, I might have some time to talk tomorrow.

Reply:
i made the recommended changes.